source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+                    tail_index        = i;
+                  }
+              }
+
+            // copy point on upper side of airfoil
+            for (unsigned int i = tail_index; i < jouk_points.size(); i++)
+              upper_points.emplace_back(jouk_points[i]);
+            for (unsigned int i = 0; i <= nose_index; i++)
+              upper_points.emplace_back(jouk_points[i]);
+            std::reverse(upper_points.begin(), upper_points.end());
+
+            // copy point on lower side of airfoil
+            for (unsigned int i = nose_index; i <= tail_index; i++)
+              lower_points.emplace_back(jouk_points[i]);
+          }
+
+          airfoil_1D[0] = make_points_equidistant(upper_points, number_points);
+          airfoil_1D[1] = make_points_equidistant(lower_points, number_points);
+
+          // move nose to origin
+          auto move_nose_to_origin = [](std::vector<Point<2>> &vector) {
+            const double nose_x_pos = vector.front()(0);
+            for (auto &i : vector)
+              i(0) -= nose_x_pos;
+          };
+
+          move_nose_to_origin(airfoil_1D[1]);
+          move_nose_to_origin(airfoil_1D[0]);
+
+          return airfoil_1D;
+        }
+
+        /**
+         * Full Joukowski circle around center point is generated beginning at
+         * most left circlepoint and then turning counterclockwise. Radius is
+         * automatically set, so that point x=-1 is enclosed by the circle and
+         * point x=1 coincides the circle.
+         *
+         *                |y
+         *          .   . |
+         *      .         | .
+         *     .          |   .
+         *    .           |    .
+         *    .       x   |    .
+         * ---.----|------|----|------>x
+         *     .  -1      |   .1
+         *      .         | .
+         *         .    . |
+         *                |
+         *
+         * @param[in] center Joukowski circle center coordinate.
+         * @param[in] number_points Number of desired circle points of full
+         * circle.
+         * @return circle_points Vector containing all circle points beginning at
+         *  point with most negative x-component, then counterclockwise.
+         */
+        static std::vector<Point<2>>
+        joukowski_circle(const Point<2> &   center,
+                         const unsigned int number_points)
+        {
+          std::vector<Point<2>> circle_points;
+
+          // Create Circle with number_points - points
+          // unsigned int number_points = 2 * points_per_side - 2;
+
+          // Calculate radius so that point (x=1|y=0) is enclosed - requirement
+          //  for Joukowski transform
+          double radius =
+            sqrt(center(1) * center(1) + (1 - center(0)) * (1 - center(0)));
+          double radius_test =
+            sqrt(center(1) * center(1) + (1 + center(0)) * (1 + center(0)));
+          // Make sure point (x=-1|y=0) is enclosed by the circle
+          AssertThrow(
```

Comment:
I suspect that you're going to get a warning about an unused variable `radius_test` in release mode. You probably want to add this:
```suggestion
          (void)radius_test;
          AssertThrow(
```